services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/DeviceDto.java

```diff
@@ -44,18 +44,16 @@ public DeviceDto() {
     /**
      * @param tenantId The tenant identifier.
      * @param deviceId The device identifier.
-     * @param device The device.
+     * @param device The device information.
      * @param version The version of tenant to be sent as request header.
-     * @param updatedOn The date and time of last update.
      * @throws NullPointerException if any of the parameters except the device are {@code null}
      */
-    public DeviceDto(final String tenantId, final String deviceId, final Device device, final long version,
-            final Instant updatedOn) {
-        this.tenantId = Objects.requireNonNull(tenantId);
-        this.deviceId = Objects.requireNonNull(deviceId);
-        this.device = device;
-        this.versionNo = version;
-        this.updatedOn = Objects.requireNonNull(updatedOn);
+    public DeviceDto(final String tenantId, final String deviceId, final Device device, final String version) {
+        setTenantId(tenantId);
+        setDeviceId(deviceId);
+        setDevice(device);
+        setVersion(version);
+        setUpdatedOn(Instant.now());
```

Comment:
are the params allowed to be `null` or not (as indicated by the JavaDoc)?

Reply:
As indicated by the _JavaDoc_, the _params_ are not allowed to be `null` except the `device`. The checks are being done in the respective _setters_. Checking for `null` again in this constructor seemed to me repetitive and hence I have removed it and then retained the `JavaDoc`.